File: jambeth/jambeth-cache/src/main/java/de/osthus/ambeth/audit/IVerifyOnLoad.java
package de.osthus.ambeth.audit;

import de.osthus.ambeth.cache.model.ILoadContainer;
import de.osthus.ambeth.collections.IList;
import de.osthus.ambeth.threading.IResultingBackgroundWorkerDelegate;

public interface IVerifyOnLoad
{
	<R> R verifyEntitiesOnLoad(IResultingBackgroundWorkerDelegate<R> runnable) throws Throwable;

	void queueVerifyEntitiesOnLoad(IList<ILoadContainer> loadedEntities);
}

File: ambeth/Ambeth.Cache/ambeth/audit/IVerifyOnLoad.cs
using De.Osthus.Ambeth.Cache.Model;
using De.Osthus.Ambeth.Threading;
using System.Collections.Generic;

namespace De.Osthus.Ambeth.Audit
{
	public interface IVerifyOnLoad
	{
		R VerifyEntitiesOnLoad<R>(IResultingBackgroundWorkerDelegate<R> runnable);

		void QueueVerifyEntitiesOnLoad(IList<ILoadContainer> loadedEntities);
	}
}